File: src/test/java/FileReaderTest.java
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertThrows;

public class FileReaderTest {

    private final FileReader fileReader;

    public FileReaderTest() {
        this.fileReader = new FileReader();
    }

    @Test
    void getWordsFromFileWithExistingFileReturnsListOfWords() throws Exception {

        int expectedResult = 4;
        List<String> result = fileReader.getWordsFromFile("stopwords.txt");

        assertEquals(expectedResult, result.size());
    }

    @Test
    void getWordsFromFileWithNonExistingFileThrowsException() {

        assertThrows(Exception.class, () -> fileReader.getWordsFromFile("badFileName.txt"));
    }
}

File: src/test/kotlin/FileReaderTest.kt
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.assertThrows
import java.io.FileNotFoundException

class FileReaderTest {
    private val fileReader = FileReader()

    @Test
    fun testReadFile() {
        val content = fileReader.readFile("src/main/resources/stopwords.txt")
        assert(content.isNotEmpty())
    }

    @Test
    fun testReadFileException() {
        assertThrows<FileNotFoundException> {
            fileReader.readFile("src/main/resources/stopwords123.txt")
        }
    }

    @Test
    fun testReadFileEmpty() {
        assertThrows<FileNotFoundException> {
            fileReader.readFile("")
        }
    }
}